### PR TITLE
Add and remove tasks from the todo list app

### PR DESCRIPTION
**Pull request by yayner2002**
in this branch, I have implemented the following features 💯  👍 👍 
in these features, you will get features that I was supposed to do for the next project. the reason is I thought it was part of the demo as seen from the video of minimalist. 🥇 

- Add task
- remove the specific task
- edit specific task
- Implement a function for the "Clear all completed" button
- text-decoration set to `line-through`
- Add event listener to the checkbox
- Update items object's value for completed key upon user actions.
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/modules/addList.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+    task.index = index;
+    toDoList.push(task);
+    localStorage.setItem('ourTasks', JSON.stringify(toDoList));
+  }
+
+  static removeTask(index) {
+    toDoList = toDoList.filter((task) => task.index !== Number(index));
+    localStorage.setItem('ourTasks', JSON.stringify(toDoList));
+  }
```
Comment (micheaol) [inline src/modules/addList.js:47]: Your app is not working as expected. I am unable to delete a task as requested by the project requirement but when I click on the `clear all task` button all the `tasks` are removed. Kindly implement `a function to remove a task` according to the project requirement.

File: src/modules/addList.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+export const listFromLocalStorage = JSON.parse(
+  localStorage.getItem('ourTasks'),
+);
+export function displayTask() {
+  // eslint-disable-next-line max-len
```
Comment (micheaol) [inline src/modules/addList.js]: Please ensure that linters are not disabled. A linter is a tool that analyzes your source code to flag programming errors, bugs, stylistic errors, and suspicious constructs(source: [Wikipedia](https://en.wikipedia.org/wiki/Lint_(software))).

File: src/modules/addList.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+const editBtn = document.querySelectorAll('.fa-pen');
+window.console.log(editBtn);
+editBtn.forEach((el) => {
+  el.addEventListener('click', (e) => {
+    const id = e.target.getAttribute('id');
+    editId = id - 1;
+    isEdited = true;
+    const name = toDoList[id - 1].description;
+    newTaskEl.value = name;
+  });
```
Comment (micheaol) [inline src/modules/addList.js]: I am unable to `edit` a task. When I click on the tasks, nothing happens. Kindly fix this and implement `edit task` according to the project requirement.

File: src/modules/addList.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    taskMenu.classList.add('show');
+    document.addEventListener('click', (e) => {
+      if (e.target.tagName !== 'I' || e.target !== menu) {
+        taskMenu.classList.remove('show');
+      }
+    });
+  });
+});
+const editBtn = document.querySelectorAll('.fa-pen');
+window.console.log(editBtn);
```
Comment (micheaol) [inline src/modules/addList.js]: Please ensure that you do not have any `console.log` in your codebase for security reasons. Kindly remove it.

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @yayner2002 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by afanimartin — APPROVED** [review]
Hi @yayner2002, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**